configure.ac

```diff
@@ -1149,6 +1159,18 @@ AC_SUBST(LEVELDB_CPPFLAGS)
 AC_SUBST(LIBLEVELDB)
 AC_SUBST(LIBMEMENV)
 
+dnl SUPPRESSED_CPPFLAGS=SUPPRESS_WARNINGS([$SOME_CPPFLAGS])
+dnl Replace -I with -isystem in $SOME_CPPFLAGS to suppress warnings from
+dnl headers from its include directories and return the result.
+dnl See -isystem documentation:
+dnl https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html
+dnl https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-isystem-directory
+dnl Do not change "-I/usr/include" to "-isystem /usr/include" because that
+dnl is not necessary (/usr/include is already a system directory) and because
+dnl it would break GCC's #include_next.
+AC_DEFUN([SUPPRESS_WARNINGS],
+         [$(echo $1 |sed -E -e 's/(^| )-I/\1-isystem /g' -e 's|-isystem /usr/include|-I/usr/include|g')])
```

Comment:
This incorrectly excludes `/usr/includefoo`

Perhaps use `s#-isystem /usr/include([/ ]|$)#-I/usr/include\1#g`

Reply:
Done, except that the `#` separator causes the following during `autogen.sh`:

```
configure.ac:11: error: m4_init: unbalanced m4_divert_push:
configure.ac:1199: m4_divert_push: GROW
configure.ac:11: m4_divert: BODY
configure.ac:1171: SUPPRESS_WARNINGS is expanded from...
configure.ac:1171: SUPPRESS_WARNINGS is expanded from...
configure.ac:1171: SUPPRESS_WARNINGS is expanded from...
configure.ac:1171: SUPPRESS_WARNINGS is expanded from...
configure.ac:1171: SUPPRESS_WARNINGS is expanded from...
configure.ac:11: the top level
autom4te-2.69: /usr/local/bin/gm4 failed with exit status: 1
aclocal: error: echo failed with exit status: 1
autoreconf-2.69: aclocal failed with exit status: 1
```

so I used `;` as a separator instead.